data/repository/src/commonMain/kotlin/social/androiddev/common/repository/timeline/TimelineRepoModule.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * This file is part of Dodo.
+ *
+ * Dodo is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * Dodo is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Dodo. If not, see <https://www.gnu.org/licenses/>.
+ */
+package social.androiddev.common.repository.timeline
+
+import com.squareup.sqldelight.runtime.coroutines.asFlow
+import com.squareup.sqldelight.runtime.coroutines.mapToList
+import kotlinx.coroutines.flow.map
+import org.koin.core.module.Module
+import org.koin.core.parameter.ParametersHolder
+import org.koin.dsl.module
+import org.mobilenativefoundation.store.store5.Fetcher
+import org.mobilenativefoundation.store.store5.SourceOfTruth
+import org.mobilenativefoundation.store.store5.StoreBuilder
+import social.androiddev.common.network.MastodonApi
+import social.androiddev.common.network.model.Status
+import social.androiddev.common.persistence.localstorage.DodoAuthStorage
+import social.androiddev.common.timeline.TimelineDatabase
+import social.androiddev.common.timeline.TimelineItem
+import social.androiddev.domain.timeline.FeedType
+import social.androiddev.domain.timeline.HomeTimelineRepository
+import social.androiddev.domain.timeline.model.StatusUI
+
+/**
+ * Koin module containing all koin/bean definitions for
+ * timeline repository delegates.
+ */
+val timelineRepoModule: Module = module {
```

Comment:
This file seems to be doing way more than just providing dependencies. The mappers should probably be moved to the right sites where they are needed and then can be easily tested

Reply:
refactored into delegates